### PR TITLE
(#1431) add hackathon id for filter

### PR DESCRIPTION
**Pull request by rz-p**

### CODE REVIEW

**Comment by rz-p** [comment]
my ideas page now will only show ideas submitted to the selected hackathon

File: hms-app/src/pages/MyIdeas.tsx [hunk 1]
```diff
@@ -61,7 +61,8 @@ export default function MyIdeas() {
 
   const filteredIdeas = relevantIdeas.filter((item) => {
     const userId = user?.id || ''
-    return item.owner?.id.includes(userId)
+    const hackathon = hackathonData.id
+    return item.owner?.id.includes(userId) && item.hackathon?.id.includes(hackathon)
```
Comment (verzano) [inline hms-app/src/pages/MyIdeas.tsx:65]: why is the check that it `includes` the id and not that it equals (`===`) the id?

Comment (rz-p) [reply hms-app/src/pages/MyIdeas.tsx:65]: Good point, will change it ASAP

File: hms-app/src/pages/MyIdeas.tsx [hunk 2]
```diff
@@ -155,6 +158,7 @@ export default function MyIdeas() {
                   columnSize={6}
                   type={IdeaCardType.Owner}
                   isLoading={false}
+                  reloadIdeaList={loadSelectedHackathon}
```
Comment (LetsPoke) [inline hms-app/src/pages/MyIdeas.tsx:161]: could you name this "OnSuccess"?

Comment (rz-p) [reply hms-app/src/pages/MyIdeas.tsx:161]: that would require me to rename the props in `IdeaCardList`
Does anyone currently working with that?

File: hms-app/src/pages/MyIdeas.tsx [hunk 3]
```diff
@@ -61,7 +61,9 @@ export default function MyIdeas() {
 
   const filteredIdeas = relevantIdeas.filter((item) => {
     const userId = user?.id || ''
-    return item.owner?.id.includes(userId)
+    return (
+      item.owner?.id === userId && item.hackathon?.id === selectedHackathonId
+    )
```
[the remark this answers is not in the document]
Comment (LetsPoke) [reply hms-app/src/pages/MyIdeas.tsx]: @verzano could you open a ticket to do a refactor like this for all checks that make sense?

[the remark this answers is not in the document]
Comment (rz-p) [reply hms-app/src/pages/MyIdeas.tsx]: Agree, would be a bit confusing if it gets changed in this PR

[the remark this answers is not in the document]
Comment (verzano) [reply hms-app/src/pages/MyIdeas.tsx]: I can, it's a bit more of a "convert types to classes" ticket though

File: hms-app/src/pages/MyIdeas.tsx (hunk 2 above)
Comment (reyhanrumengan) [reply hms-app/src/pages/MyIdeas.tsx]: @rz-p you change the props to `onSuccess`, but haven't change it in the `IdeaCardList`(?)
Then you also need to change it in a few components that uses `IdeaCardList`

File: hms-app/src/pages/MyIdeas.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rz-p) [reply hms-app/src/pages/MyIdeas.tsx]: added ticket #1439 for this issue. Feel free to add things to the ticket @verzano

**Pull request merged**